### PR TITLE
Add Django-project example

### LINKED ISSUE #9: Django example
Can you give some pointers for creating a package around Django? I would be happy to test, verify and PR a working example of the same, but could use some pointers on it if you can advise?

To clarify I am looking to create a package like https://github.com/datakurre/setup.nix/tree/master/examples/package that I can include in a nixos iso build I am working on

### PR DESCRIPTION
**Pull request by datakurre**
fixes #9 

@samrose

```
make nix-build
result/bin/manage runserver --noreload
```

or just

```
make runserver
```

Nixpkgs Python-script wrapping breaks manage script without "--noreload", but it works properly in Nix-shell ("make nix-shell").

### CODE REVIEW

File: examples/django/src/django_project/manage.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        raise ImportError(
+            "Couldn't import Django. Are you sure it's installed and "
+            "available on your PYTHONPATH environment variable? Did you "
+            "forget to activate a virtual environment?"
+        ) from exc
+    execute_from_command_line(sys.argv)
+
+
+if __name__ == '__main__':
+    main()
```
Comment (datakurre) [inline examples/django/src/django_project/manage.py]: @samrose This "django_project"-module was created with django-admin by following https://dfpp.readthedocs.io/en/latest/chapter_01.html

I made the project into a Python package and refactored manage.py-script into a entry-point script in that package.

File: examples/django/src/django_project/settings.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+WSGI_APPLICATION = 'django_project.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/2.1/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': os.path.join(os.getcwd(), 'db.sqlite3'),
```
Comment (datakurre) [inline examples/django/src/django_project/settings.py]: @samrose By default this pointed into a path that become readonly when package is built with Nix. Therefore I changed this to point to current working directory.

**Comment by samrose** [comment]
@datakurre thank you so much this all makes sense. I'll be working with it today

**Comment by samrose** [comment]
@datakurre I just did a simple test on this and it worked beautifully. Thank you so much

can this approach also handle the gunicorn serving of a django application?

**Comment by datakurre** [comment]
@samrose I assume that the “best practise” for that would be to create another derivation composing a Python environment with both gunicorn and this django project package so that gunicorn could be called with django project on its Python path.

**Comment by datakurre** [comment]
@samrose 

```bash
nix-build gunicorn.nix
result/bin/gunicorn django_project.wsgi
```

**Comment by datakurre** [comment]
```
{ pkgs ? import <nixpkgs> {}
, pythonPackages ? pkgs.python3Packages
}:

pythonPackages.python.withPackages (ps: with ps; [
  gunicorn
  (import ./setup.nix {
    inherit pkgs pythonPackages;
  }).build
])
```

**Comment by datakurre** [comment]
I don't remember anymore, why i didn't make that `.build` the default result of that function. Now it just returns attrset with attributes like build, sdist, bdist_wheel, env, shell...

Maybe I considered that too magical.

**Comment by samrose** [comment]
@datakurre anything I can do to help document this? it may be the example is enough? Thanks again for creating it!

**Comment by datakurre** [comment]
@samrose Well, if you have time, you could ensure that this still works and maybe add a small README into the example folder.

But also the top-level README is now outdated. See https://github.com/datakurre/setup.nix/issues/13

Some time ago, I had to fix this to work also with NixOS 18.10 (pip >= 18). For that I also had to fork pip2nix that is still missing NixOS 18.10 support (besides my fork). Everything should work, but I have not managed to update README nor the requirements.nix-files in examples (and that's why smoketests at Travis-CI are failing).

But I assume I'll manage to fix README by the next week this time. Meanwhile, i'm figuring out support for private repositories...

**Pull request merged**